packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -57,37 +54,21 @@ export const createPluginActions: CreatePluginActions = (
   //  Because the container doesn't send the full state in its messages, we need to track it ourselves.
   //  isModal and height is not included in the messages to the children and must thus be tracked here.
   //  In future improved versions of the plugin API, this should not be needed.
-  let state: FieldPluginData = defaultState
 
-  let assetSelectedCallbackRef: undefined | ((filename: Asset) => void) =
-    undefined
-  let assetSelectedCallbackId: undefined | string = undefined
+  const { pushCallback, popCallback } = callbackQueue()
```

Comment:
This function returns two functions that lets you manage callback functions. 

When pushing, you specify the type of the callback function, and the callback function itself. It returns an id that you can include in the message to the visual editor.

When popping, you specify the callbackId and the type of the callback function, this way, typescript is able to guarantee that the callback function is of the expected type.